ChainSafe/forest
# fix: don't GC manifests

## Summary of changes



Changes introduced in this pull request:

- blessed column for persistent storage

## Reference issue to close (if applicable)





Closes #4926 

## Other information and links

The actors are found fine after several GC rounds, so this is confirmed. Also, the migration now works as intended.

Manual testing:

Several GC runs prior to state migration:

```
2024-11-12T14:19:05.975826Z  INFO forest_filecoin::db::gc: filter keys for GC
2024-11-12T14:19:06.544906Z  INFO forest_filecoin::db::gc: GC sweep
2024-11-12T14:19:06.793863Z  INFO forest_filecoin::db::gc: GC finished sweep: 4189 deleted records
```


There are of course other errors that have to do with modifying the state migration epoch, but no bundle retrieval issues, I have also tested this before the `persistent` fix and it failed miserably trying to retrieve those from the DB.
```
2024-11-12T15:18:29.939559Z  INFO compute_tipset_state_blocking: forest_filecoin::state_migration: Running TukTuk migration at epoch 2136520
2024-11-12T15:18:29.985712Z  INFO forest_filecoin::chain_sync::tipset_syncer: Validating tipset: EPOCH = 2136611, N blocks = 1
2024-11-12T15:18:29.985734Z  WARN forest_filecoin::chain_sync::tipset_syncer: Got block from the future, but within clock drift threshold, 1731424710 > 1731424709
2024-11-12T15:18:29.988647Z  WARN forest_filecoin::chain_sync::tipset_syncer: Validating block [CID = bafy2bzacea3gbxtureg5u7msimtsnns2j7edqmsetgghs7uy4cxlqdnejgzgo] in EPOCH = 2136611 failed: Validation error: Processing error: Error calculating weight: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Processing error: Failed to calculate state: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Processing error: Could not update state: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Validation error: Consensus error: Power actor not found, Consensus error: StateManager error: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown
2024-11-12T15:18:29.988693Z  WARN forest_filecoin::chain_sync::tipset_syncer: Sync messages check state failed for single tipset
2024-11-12T15:18:29.988701Z ERROR forest_filecoin::chain_sync::tipset_syncer: Syncing tipset range [2136521, 2136611] failed: Validation error: Processing error: Error calculating weight: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Processing error: Failed to calculate state: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Processing error: Could not update state: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown, Validation error: Consensus error: Power actor not found, Consensus error: StateManager error: Can't create a valid state tree from the given root. This error may indicate unsupported version. state_root_cid=bafy2bzaceaxzciqzroykafdbt2fcizvzphu2hex4dceseo3lzrbl6qtuy3d7u, state_root_version=unknown
thread 'state migration thread: 0' panicked at /Users/romanuseinov/projects/chainsafe/forest/src/state_migration/common/state_migration.rs:134:29:
failed executing job for address: t04, Reason: state migration failed for bafk2bzaceclefusmffhuuvtggrmadr3cwpwsgphtlj2wb222ztwwv5mssu5ea actor, addr t04:Serialization error for Cbor protocol: RequireLength { name: "tuple", expect: 15, value: 17 }
2024-11-12T15:18:30.090031Z  INFO forest_filecoin::state_migration::common::state_migration: Processed 100000 actors
```

## Change checklist




- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.



[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md
